src/Functions/array/arrayAggregation.cpp

```diff
@@ -360,7 +360,7 @@ struct ArrayAggregateImpl
 
     static ColumnPtr execute(const ColumnArray & array, ColumnPtr mapped)
     {
-        if constexpr (aggregate_operation == AggregateOperation::max || aggregate_operation == AggregateOperation::min)
+        if (!mapped->isNumeric() && !isNumber(array.getDataPtr()->getDataType()) && (aggregate_operation == AggregateOperation::max || aggregate_operation == AggregateOperation::min))
```

Comment:
I don't know the inners of the logic, but I suggest to put the `!mapped->isNumeric()&& ...` inside the `if constexpr (...` body block.

The `if constexpr` is evaluated at compile time, but the `!mapped->isNumeric() && ...` is evaluated at runtime.